src/Microsoft.AspNetCore.DataProtection.Abstractions/DataProtectionExtensions.cs

```diff
@@ -120,15 +121,19 @@ public static IDataProtector CreateProtector(this IDataProtectionProvider provid
         [EditorBrowsable(EditorBrowsableState.Never)]
         public static string GetApplicationUniqueIdentifier(this IServiceProvider services)
         {
-            string discriminator = (services?.GetService(typeof(IApplicationDiscriminator)) as IApplicationDiscriminator)?.Discriminator;
-            if (discriminator == null)
+            string discriminator = null;
```

Comment:
Should this method be in Microsoft.AspNetCore.DataProtection so the .Abstractions project doesn't need a dependency on Hosting? That dependency seems unnecessary and undesirable for the Abstractions project, especially given that it is only used here.